### PR TITLE
gnrc_ipv6_whitelist: initial import

### PR DESCRIPTION
**Pull request by authmillenon**
This provides the capability to emulate non-reachability between nodes. This can be helpful for testing routing protocols e.g.

### CODE REVIEW

File: sys/shell/commands/shell_commands.c [hunk 1]
```diff
@@ -196,6 +200,9 @@ const shell_command_t _shell_command_list[] = {
     {"ncache", "manage neighbor cache by hand", _ipv6_nc_manage },
     {"routers", "IPv6 default router list", _ipv6_nc_routers },
 #endif
+#ifdef MODULE_GNRC_IPV6_WHITELIST
+    {"whitelist", "whitelists an address for receival ('whitelist [add|del|help]')", _whitelist },
```
Comment (cgundogan) [inline sys/shell/commands/shell_commands.c]: ... `an IPv6 address` ...

Comment (authmillenon) [inline sys/shell/commands/shell_commands.c]: Why, the thought was, that it might be expanded for IPv4 (or link-layer) some day :-)

Comment (cgundogan) [inline sys/shell/commands/shell_commands.c]: Not as long as the module itself has `IPv6` in its name (:

Comment (authmillenon) [inline sys/shell/commands/shell_commands.c]: This is a shell command, not a module ;-)

Comment (authmillenon) [inline sys/shell/commands/shell_commands.c]: The idea was, that the `whitelist` shell command can be from the outside stay more or less the same, while what it does might change.

**Comment by cgundogan** [comment]
tested on native and on iotlab testbed. I was not able to ping nodes if they were not whitelisted. ACK

**Comment by authmillenon** [comment]
Squashed.

**Comment by cgundogan** [comment]
travis is happy

**Comment by cgundogan** [comment]
BTW: I tested with `USEMODULE+=gnrc_ipv6_whitelist make -B clean all-debug debug PORT=tap0`

**Pull request merged**